cosmos/airflow/graph.py

```diff
@@ -111,7 +111,7 @@ def calculate_leaves(tasks_ids: list[str], nodes: dict[str, DbtNode]) -> list[st
 def exclude_detached_tests_if_needed(
     node: DbtNode,
     task_args: dict[str, str],
-    detached_from_parent: dict[str, DbtNode] | None = None,
+    detached_from_parent: dict[str, list[DbtNode]] | None = None,
```

Comment:
If we want this change to be non-breaking:
```suggestion
    detached_from_parent: dict[str, DbtNode | list[DbtNode]] | None = None,
```
And we can normalise within the function itself:
```
        if detached_from_parent is None:
            detached_from_parent = {}
        else:
            # Normalize so that every value is always a list
            detached_from_parent = {
                key: value if isinstance(value, list) else [value]
                for key, value in detached_from_parent.items()
            }
```